admin/app/controllers/solidus_admin/backend_auth_support.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module SolidusAdmin::BackendAuthSupport
+  extend ActiveSupport::Concern
+
+  private
+
+  def authenticate_solidus_backend_user!
+    if respond_to?(:model_class, true) && model_class
+      record = model_class
+    else
+      record = controller_name.to_sym
+    end
+    authorize! :admin, record
+    authorize! action_name.to_sym, record
+  rescue CanCan::AccessDenied
+    instance_exec(&Spree::Admin::BaseController.unauthorized_redirect)
+  end
+
+  # Needs to be overriden so that we use Spree's Ability rather than anyone else's.
+  def current_ability
+    @current_ability ||= Spree::Ability.new(spree_current_user)
+  end
+
+  def store_location
```

Comment:
Where is this method used? :thinking:

Reply:
I see. Do you think it's worth it to drop a comment so it's clear for those reading the code?